src/Microsoft.ServiceFabric.Services.Remoting/Description/MethodDescription.cs

```diff
@@ -61,10 +61,11 @@ public MethodInfo MethodInfo
      
         internal static MethodDescription Create(string remotedInterfaceKindName, MethodInfo methodInfo)
         {
-            var argumentList = new List<MethodArgumentDescription>();
+            var parameters = methodInfo.GetParameters();
+            var argumentList = new List<MethodArgumentDescription>(parameters.Length);
             var hasCancellationToken = false;
 
-            foreach (var param in methodInfo.GetParameters())
+            foreach (var param in parameters)
```

Comment:
those changes are for InterfaceDescription.cs, the changes mentioned above should be in MethodDescription.cs. Chnages in MethodDescription.cs is moving methodInfo.GetParameters() before the loop which doesn't help with perf as compiler moves it outside of the loop as optimization.